File: storage/data_transpiled/at_coder/abc081/B/4927229.java
public static int N ( ) {
  int [ ] A = new int [ N ] ;
  for ( int i = 0 ;
  i < N ;
  i ++ ) {
    A [ i ] = Integer . parseInt ( input . nextLine ( ) ) ;
  }
  int ans = 0 ;
  boolean k = true ;
  while ( k == true ) {
    for ( int i = 0 ;
    i < A . length ;
    i ++ ) {
      if ( A [ i ] % 2 == 1 ) {
        k = false ;
        break ;
      }
    }
    if ( k == true ) {
      ans ++ ;
      for ( int i = 0 ;
      i < A . length ;
      i ++ ) {
        A [ i ] = A [ i ] / 2 ;
      }
    }
  }
  return ans ;
}

File: storage/data_original/at_coder/abc081/B/4927229.py
N = int(input())
A = list(map(int, input().split()))
ans = 0
k = True

while k == True:
	for i in range(len(A)):
		if A[i] % 2 == 1:
			k = False
			break

	if k == True:
		ans += 1
		for i in range(len(A)):
			A[i] = A[i] / 2

print(ans)